### ISSUE TITLE
Encoding on encoded url for image

### ISSUE THREAD

**Issue by santoshgistto** [swissspidy/web-stories-wp#11334]
## Bug Description
Image url  is getting encoded without checking whether the url is already encoded while adding image url in the element. 

Note: This may not be a bug  for wordpress case

## Expected Behaviour
Would be good if we check the url before encoding.

## Steps to Reproduce
`Used storybook`
1. Use firebase storage image url 
2. Add image url in `getDummyMedia` list 
3. Go to  stories editor and add image from media section to the story card.

For more details please see below video 

Sample : 
```
    createDummyImageData({
      mimeType: 'image/jpeg',
      src: 'https://firebasestorage.googleapis.com/v0/b/c-dashboard-d4a82.appspot.com/o/media%2FNurUn5ekSeOkSCfk1yPIkg17buI3%2Fimages%2F1650068451121-omid-armin-nACf6L_pXq8-unsplash.jpeg?alt=media&token=edc4dfd7-6ac1-44a0-83b7-1aa99a3adad3',
      width: 5262,
      height: 7981,
      id: 1,
      title: 'dummy image',
    })
```
Actual image url : 
```
https://firebasestorage.googleapis.com/v0/b/c-dashboard-d4a82.appspot.com/o/media%2FNurUn5ekSeOkSCfk1yPIkg17buI3%2Fimages%2F1650068451121-omid-armin-nACf6L_pXq8-unsplash.jpeg?alt=media&token=edc4dfd7-6ac1-44a0-83b7-1aa99a3adad3
```

Encoded srcSet url : 
```
https://firebasestorage.googleapis.com/v0/b/c-dashboard-d4a82.appspot.com/o/media%252FNurUn5ekSeOkSCfk1yPIkg17buI3%252Fimages%252F1650068451121-omid-armin-nACf6L_pXq8-unsplash.jpeg?alt=media&token=edc4dfd7-6ac1-44a0-83b7-1aa99a3adad3 412w
```

## Screenshots
<img width="308" alt="Screen Shot 2022-04-21 at 11 01 29 PM" src="https://user-images.githubusercontent.com/12577046/164587886-f006c813-2c72-49cc-becd-ebe85832f19d.png">

## Video
https://user-images.githubusercontent.com/12577046/164589464-08e291c7-f0b3-4ea8-a918-0ba55045a78c.mov

## Additional Context
https://github.com/GoogleForCreators/web-stories-wp/blob/925e8057d693f05c3803740d8634617f2825ce7b/packages/media/src/calculateSrcSet.js#L50

- Branch : main branch 
- WordPress Version: Not applicable 
- Operating System: mac
- Browser: chrome

**Comment by swissspidy** [comment]
Thanks for reporting!

Sounds like we ought to be able to make this more robust.

For context, this was introduced in #9452

**Comment by spacedmonkey** [comment]
@swissspidy Can you provide more detail on what is needs to be done here?

**Comment by swissspidy** [comment]
Sure, I can do that. Maybe even easiest for me to pick this one up.

In the meantime we have a few P1s that are more urgent.

**Comment by swissspidy** [comment]
@santoshgistto I do have a possible fix for this ready at #11731. Does that resolve the issue for you?

**Comment by santoshgistto** [comment]
Sorry for late response @swissspidy 
https://github.com/GoogleForCreators/web-stories-wp/pull/11731 its working thank you 💯
